gazebo_plugins/src/gazebo_ros_bumper.cpp

```diff
@@ -0,0 +1,226 @@
+  gazebo_ros::Node::SharedPtr ros_node_{nullptr};
+
+  /// Contact mesage publisher.
+  rclcpp::Publisher<gazebo_msgs::msg::ContactsState>::SharedPtr pub_{nullptr};
+
+  /// Pointer to sensor
+  gazebo::sensors::ContactSensorPtr parent_sensor_;
+
+  /// Frame name, to be used by TF.
+  std::string frame_name_;
+
+  /// Connects to pre-render events.
+  gazebo::event::ConnectionPtr update_connection_;
+};
+
+GazeboRosBumper::GazeboRosBumper()
+: impl_(std::make_unique<GazeboRosBumperPrivate>())
+{
+}
+
+GazeboRosBumper::~GazeboRosBumper()
+{
+  impl_->ros_node_.reset();
+}
+
+void GazeboRosBumper::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _sdf)
+{
+  // Initialize ROS node
+  impl_->ros_node_ = gazebo_ros::Node::Get(_sdf);
+
+  impl_->parent_sensor_ = std::dynamic_pointer_cast<gazebo::sensors::ContactSensor>(_sensor);
+  if (!impl_->parent_sensor_) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+      "Contact sensor parent is not of type ContactSensor. Aborting");
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  // Contact state publisher
+  impl_->pub_ = impl_->ros_node_->create_publisher<gazebo_msgs::msg::ContactsState>(
+    "bumper_states", rclcpp::SensorDataQoS());
+
+  RCLCPP_INFO(impl_->ros_node_->get_logger(), "Publishing contact states to [%s]",
+    impl_->pub_->get_topic_name());
+
+  // Get tf frame for output
+  impl_->frame_name_ = _sdf->Get<std::string>("frame_name", "world").first;
+
+  impl_->update_connection_ = impl_->parent_sensor_->ConnectUpdated(
+    std::bind(&GazeboRosBumperPrivate::OnUpdate, impl_.get()));
+
+  impl_->parent_sensor_->SetActive(true);
+}
+
+void GazeboRosBumperPrivate::OnUpdate()
+{
+  gazebo::msgs::Contacts contacts;
+  contacts = parent_sensor_->Contacts();
+
+  gazebo_msgs::msg::ContactsState contact_state_msg;
+  contact_state_msg.header.frame_id = frame_name_;
+  contact_state_msg.header.stamp =
+    rclcpp::Time(contacts.time().sec(), static_cast<uint32_t>(contacts.time().nsec()));
+
+  ignition::math::Pose3d pose, frame_pose;
+  ignition::math::Quaterniond rot, frame_rot;
+  ignition::math::Vector3d pos, frame_pos;
+
+  frame_pos = ignition::math::Vector3d(0, 0, 0);
+  frame_rot = ignition::math::Quaterniond(1, 0, 0, 0);  // gazebo u,x,y,z == identity
+  frame_pose = ignition::math::Pose3d(frame_pos, frame_rot);
+
+  // set contact states size
+  contact_state_msg.states.clear();
+
+  // GetContacts returns all contacts on the collision body
+  int contactsPacketSize = contacts.contact_size();
+  for (int i = 0; i < contactsPacketSize; ++i) {
+    // For each collision contact
+    // Create a ContactState
+    gazebo_msgs::msg::ContactState state;
+    const gazebo::msgs::Contact & contact = contacts.contact(i);
+
+    state.collision1_name = contact.collision1();
+    state.collision2_name = contact.collision2();
+    std::ostringstream stream;
+    stream << "Debug:  i:(" << i << "/" << contactsPacketSize <<
+      ")     my geom:" << state.collision1_name <<
```

Comment:
Nitpick: add a space after `:`s. Or to make it even clearer, wrap the collision names on `[]` so they're easy to spot. Right now the message is a bit hard to read: `Debug:  i:(0/1)     my geom:ball::link::collision   other geom:ground_plane::link::collision    
     time:12.862000000`.

Also, `0/1` looks a bit weird if it's the only message. Maybe print `i + 1`?